pkg/sm/sm.go

```diff
@@ -155,6 +155,9 @@ func (smb *ServiceManagerBuilder) Build() *ServiceManager {
 }
 
 func initializeSecureStorage(secureStorage storage.Security) error {
+	if err := secureStorage.Lock(); err != nil {
+		return err
```

Comment:
We don't want to return an error if database is locked. I think it should be:
1) Try to lock database.
2) If it is already locked, then wait some time (2 seconds for example)
3) Try to lock again
4) Repeat several times(5 for example)
5) If it couldn't acquire the lock, then panic
6) If lock is acquired then check for the encryption key. If it is already there then do nothing, otherwise generate one

Reply:
It appears that advisory locks block until a lock can be acquired: 
https://medium.com/@codeflows/application-level-locking-with-postgres-advisory-locks-c29759eeb7a7